data/transactions/logic/eval.go

```diff
@@ -252,6 +251,9 @@ type EvalParams struct {
 
 	// Total pool of app call budget in a group transaction
 	PooledApplicationBudget *uint64
+
+	// A timestamp of the Txn's (FirstValid - 1) block
+	FirstValidTimestamp uint64
```

Comment:
Should timestamp be `uint64` or `int64`?

Reply:
All our teal code is `uint64`, so I'd think we would want to keep this as uint64. Plus, all our timestamps are always positive numbers.